src/backend/distributed/commands/table.c

```diff
@@ -1070,6 +1077,30 @@ AlterTableCommandTypeIsTrigger(AlterTableType alterTableType)
 }
 
 
+/*
+ * DropsForeignKey returns true if the given AlterTableStmt drops a foreign key
+ * for the given relationId. False otherwise.
+ */
+static bool
+DropsForeignKey(Oid relationId, AlterTableStmt *alterTableStatement)
+{
+	if (list_length(alterTableStatement->cmds) != 1)
+	{
+		return false;
+	}
+
+	AlterTableCmd *command = linitial(alterTableStatement->cmds);
+	AlterTableType alterTableType = command->subtype;
+	if (alterTableType != AT_DropConstraint)
+	{
+		return false;
+	}
+
+	char *constraintName = command->name;
+	return ConstraintIsAForeignKey(constraintName, relationId);
```

Comment:
Regarding https://github.com/citusdata/citus/pull/4674#pullrequestreview-587366278, I know we decided not to cover every possible ddl in this pr, but I think we need to make sure that we consider below cases:

i) alter table drops foreign key
ii) alter table drops a column in the either side of the foreign key
iii) alter table drops a unique/primary constraint that would cascade to a foreign key

-----

For i):

As @onderkalaci also mentioned:
>I was expecting this check to be in the case where referencingIsLocalTable is local, all other cases should have been already covered:

Alter table commands defining foreign keys on local tables does not go through post process, that is why we don't consider drop foreign key commands for local tables but citus tables (`postprocess->ErrorIfUnsupportedAlterTableStmt->MarkInvalidateForeignKeyGraph`)

With the current changes made in this pr, we invalidate `drop foreign key` commands for local tables too.

----

For ii) & iii), citus was already not invalidating fkey graph for those commands, even for citus tables.
Maybe we should re-use db5bae8ffa27d363f23dc63f6d5afe365f961962 & f29d7f19830e8cdcf7169bbc720b56eb3c7003ac commits directly in this pr ?